llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -3855,11 +3855,19 @@ bool RISCVDAGToDAGISel::performCombineVMergeAndVOps(SDNode *N) {
   // If we end up changing the VL or mask of True, then we need to make sure it
   // doesn't raise any observable fp exceptions, since changing the active
   // elements will affect how fflags is set.
-  if (TrueVL != VL || !IsMasked)
+  if (TrueVL != VL || !IsMasked) {
     if (mayRaiseFPException(True.getNode()) &&
         !True->getFlags().hasNoFPExcept())
       return false;
 
+    // If the EEW of True is different from vmerge's SEW, then we cannot change
+    // the VL or mask.
+    if (Log2_64(True.getScalarValueSizeInBits()) !=
```

Comment:
Maybe don't use log here, we do a shift left to `log2sew` instead. Shift is cheaper than log I think.